controlplane/controller/cmd/cdiff/entrypoint.sh

```diff
@@ -0,0 +1,78 @@
+#!/bin/bash
+set -euo pipefail
+
+
+# Usage: entrypoint.sh [-v|--verbose] <environment> <current_controller_addr>
+VERBOSE=0
+if [[ "$1" == "-v" || "$1" == "--verbose" ]]; then
+  VERBOSE=1
+  shift
+fi
+
+if [ "$#" -ne 2 ]; then
+  echo "Usage: $0 [-v|--verbose] <environment> <current_controller_addr>"
+  exit 1
+fi
+
+OUTDIR="/var/tmp/devices_out"
+ENV="$1"
+CURR_ADDR="$2"
+
+mkdir -p "$OUTDIR"
+
+cd "$OUTDIR"
+if [ ! -d .git ]; then
+  git init > /dev/null 2>&1
+fi
+cd - > /dev/null
+
+case "$ENV" in
+  devnet|testnet|mainnet)
+    ;;
+  *)
+    echo "Unknown environment: $ENV"
+    exit 1
+    ;;
+esac
+
+if [ "$VERBOSE" -eq 1 ]; then
+  /app/cdiff -out "$OUTDIR" -env "$ENV" -controller "$CURR_ADDR"
+else
+  if ! /app/cdiff -out "$OUTDIR" -env "$ENV" -controller "$CURR_ADDR" > /dev/null 2>&1; then
+    echo "Error: cdiff failed for current controller ($CURR_ADDR)"
+    exit 1
+  fi
+fi
+
+cd "$OUTDIR"
+git add . > /dev/null 2>&1
+git commit -m "current production configs" > /dev/null 2>&1 || true
+cd - > /dev/null
+
+if [ "$VERBOSE" -eq 1 ]; then
+  /app/controller start -enable-interfaces-and-peers -env "$ENV" -listen-port 7000 &
```

Comment:
Can remove `-enable-interfaces-and-peers` since https://github.com/malbeclabs/doublezero/pull/1983 was merged